File: credit.c
#include <stdio.h>
#include <cs50.h>

bool checkIfValid(long input);
int getLength(long input);
bool checkSum(long input);
void checkforBrandAndPrint(long input);

int main(void) 
{
    long input = 0;
    do 
    {
        input = get_long("Number: \n");
    }
    while (input < 0);
    
    if (checkIfValid(input))
    {
        checkforBrandAndPrint(input);
    }
    else 
    {
        printf("INVALID\n");
    }
}

bool checkIfValid(long input)
{
    int len = getLength(input);
    return (len == 13 || len == 15 || len == 16) && checkSum(input);
}

int getLength(long input)
{
    int l;
    for (l = 0; input != 0; input /= 10, l++);
    return l;
}

bool checkSum(long input)
{
    int sum = 0;
    for (int p = 0; input != 0; input /= 10, p++)
    {
        if (p % 2 == 0)
        {
            sum += input % 10;
        }
        else
        {
            int digit = 2 * (input % 10);
            sum += (digit / 10) + (digit % 10);
        }
    }
    return (sum % 10) == 0;
}

void checkforBrandAndPrint(long input)
{
    string print; 
    if ((input >= 34e13 && input < 35e13) || (input >= 37e13 && input < 38e13))
    {
        print = "AMEX\n";
    }
    else if (input >= 51e14 && input < 56e14)
    {
        print = "MASTERCARD\n";
    }
    else if ((input >= 4e12 && input < 5e12) || (input >= 4e15 && input < 5e15))
    {
        print = "VISA\n";
    }
    else
    {
        print = "INVALID\n";
    }
    printf("%s", print);
}

File: credit.py
from cs50 import get_int


def main():
    while True:
        inpt = get_int("Number: \n")
        if inpt > 0:
            break

    if check_if_valid(inpt):
        check_for_brand_and_print(inpt)
    else:
        print("INVALID")


def check_if_valid(inpt):
    lngth = len(str(inpt))
    return (lngth in [13, 15, 16]) and check_sum(inpt)


def check_sum(inpt):
    p = 0
    summ = int(0)
    while int(inpt) != 0:
        if p % 2 == 0:
            summ += int(inpt % 10)
        else:
            digit = int(2 * (int(inpt) % 10))
            summ += int((digit / 10) + (digit % 10))
        p += 1
        inpt /= 10
    return (int(summ) % 10) == 0


def check_for_brand_and_print(ipt):
    inpt = int(ipt)
    if 34e13 <= inpt < 35e13 or 37e13 <= inpt < 38e13:
        brand = "AMEX"
    elif 51e14 <= inpt < 56e14:
        brand = "MASTERCARD"
    elif 4e12 <= inpt < 5e12 or 4e15 <= inpt < 5e15:
        brand = "VISA"
    else:
        brand = "INVALID"
    print(brand)


main()